readthedocs/doc_builder/environments.py

```diff
@@ -726,10 +727,18 @@ def __init__(self, *args, **kwargs):
                 project_name=self.project.slug,
             )[:DOCKER_HOSTNAME_MAX_LEN],
         )
+
+        # Decide what Docker image to use, based on priorities:
+        # Use the Docker image set by user or,
         if self.config and self.config.build.image:
             self.container_image = self.config.build.image
+        # the image overridden by the project (manually set by an admin) or,
         if self.project.container_image:
             self.container_image = self.project.container_image
+        # ``testing`` image if the project has this feature flag
+        if self.project.has_feature(Feature.USE_TESTING_BUILD_IMAGE):
+            self.container_image = 'readthedocs/build:testing'
```

Comment:
Shouldn't this be _first_ so that it doesn't overwrite the users config?

Reply:
I think the idea was to override the users config

Reply:
I think if we're trying to test things, we should do it on users that haven't explicitly expressed a preference. It feels wrong to override them, and more likely to break something.

Reply:
My idea is to override this user setting, yes. I explained my opinion better at https://github.com/rtfd/readthedocs.org/pull/5109#issuecomment-454473316